source/github-checks.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+const chalk = require('chalk');
+const delay = require('delay');
+const ghGot = require('gh-got');
+const pTimeout = require('p-timeout');
+const pWaitFor = require('p-wait-for');
+const terminalLink = require('terminal-link');
+const git = require('./git-util');
+const util = require('./util');
+
+const THIRTY_SECONDS = 1000 * 30;
+const THIRTY_MINUTES = 1000 * 60 * 30;
+
+module.exports = async (task, pkg, options) => {
+	const {title} = task;
+
+	if (await git.hasUnpushedCommits()) {
+		if (options.preview) {
+			return task.skip('[Preview] Will not ensure checks have passed because there are unpushed commits.');
+		}
+
+		await git.push();
+
+		// It might take a bit for the status checks to be created.
+		// Even if, it probably takes at least a few seconds for them to pass.
+		task.title = `${title} ${chalk.yellow('(waiting for checks to start…)')}`;
+		await delay(THIRTY_SECONDS);
```

Comment:
Wouldn't 10-15s suffice in this case?
Also, is this really needed? Won't `pWaitFor` do this for us anyways?

Reply:
> Wouldn't 10-15s suffice in this case?

Probably. I just sticked to the existing delay to keep it simple. If there are actually any status checks, I’d expect them to take more than 30 seconds to complete for most projects anyway.

> Also, is this really needed? Won't `pWaitFor` do this for us anyways?

If we immediately call GitHub’s status API, `total_count` might still be zero and we’d skip the task with “No status checks found”. On the flip side, we can’t retry in that case because commits without any status checks would be caught in a loop then (since the `state` will always be `pending`).

Reply:
> Wouldn't 10-15s suffice in this case?

I agree. 30 seconds is too long.

> It might take a bit for the status checks to be created.

This sounds like a GitHub bug. Why wouldn't they be created right away?

Reply:
> I agree. 30 seconds is too long.

Changed to 10 seconds in c704b6cc2b19037912cf7650b7a3dbfc1961f59c.

> This sounds like a GitHub bug. Why wouldn't they be created right away?

As far as I understand, GitHub can’t know whether a commit will result in status checks being created. For example, GitHub only notifies Travis CI of Git pushes and it’s then Travis CI’s call whether to create any status checks (which it might not need to, depending on your settings).

Reply:
Good point. What about GitHub Actions? I assume GitHub would know then?

Reply:
GitHub could but I still think there would be a slight delay. [According to GitHub Help](https://help.github.com/en/actions/configuring-and-managing-workflows/managing-a-workflow-run), GitHub Actions uses the [Checks API](https://developer.github.com/v3/checks/) to output statuses. Therefore, I think the Statuses API would still return zero status checks in the time frame between the commit having been pushed and the workflow starting to run.